### PR TITLE
Release of version 8.1.0

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Preparation for release 8.1.0
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by s3tezsky** [comment]
Original dump of CHANGELOG

```
<!-- dumped content start -->

## Unreleased

### Unknown Package

#### Changed

- [#1487] Improve working with currency rounding and fraction digits, Thanks to [@malyMiso]
- [#1364] ProductCodeController now autowires PromoCodeDataFactoryInterface instead of PromoCodeDataFactory, Thanks to [@s3tezsky]
- [#1360] disallow administrator login with default credentials in production e…, Thanks to [@s3tezsky]

#### Fixed

- [#1481] Fixed setting user permissions for travis, Thanks to [@s3tezsky]

#### Removed

- [#1422] Readme update to current state and removed package architecture image

#### Unknown Category

- [#1524] preserve custom checkbox CSS classes, Thanks to [@malyMiso]
- [#1523] update upgrade notes for currencies, Thanks to [@malyMiso]

### docs

#### Fixed

- [#1533] fix upgrade notes 8.1, Thanks to [@s3tezsky]
- [#1528] fixed upgrade notes phpunit injector, Thanks to [@s3tezsky]
- [#1526] fixed upgrade notes for cleaning admin styles, Thanks to [@s3tezsky]

#### Unknown Category

- [#1500] upgrade notes for styleguide, Thanks to [@TomasGottvald]
- [#1537] update upgrade notes, Thanks to [@TomasGottvald]

### [shopsys/framework]

#### Added

- [#1517] added missing translations for promo codes

#### Changed

- [#1452] design - admin - sorting arrows not changing column width, Thanks to [@TomasGottvald]
- [#1314] CronModuleExecutor is now service in DIC, Thanks to [@pk16011990]

#### Fixed

- [#1399] Admin/LoginController: fixed dumping translation, Thanks to [@s3tezsky]
- [#1441] fixed image drag&drop in product gallery, Thanks to [@TomasGottvald]

#### Unknown Category

- [#1426] todo.less is incorporated properly into framework, Thanks to [@TomasGottvald]
- [#1560] message error with long words, Thanks to [@TomasGottvald]
- [#1401] AdministratorChecker: verify default password after regular authentication
- [#1386] AdministratorChecker uses environment from DIC parameter
- [#1361] use DIC configuration instead of factory class to create Redis caches
- [#1527] exclude styleguide plugins from dumping translations
- [#1340] replace constants from other classes by literal values, Thanks to [@sspooky13]
- [#1446] refactoring error messages at admin order detail form, Thanks to [@TomasGottvald]

### [shopsys/http-smoke-testing]

- [#1423] HttpSmokeTestCase uses constants for kernel options

### [shopsys/project-base]

#### Added

- [#1342] added travis build for automated testing of installation script to project-base, Thanks to [@s3tezsky]
- [#1561] - error flash messages - add correct type to template, Thanks to [@TomasGottvald]
- [#1465] Add graphic checkboxes and radiobuttons, Thanks to [@TomasGottvald]
- [#1472] - add posibility to override admin styles, Thanks to [@TomasGottvald]
- [#1483] added support for image and iframe lazy loading, Thanks to [@TomasGottvald]

#### Changed

- [#1330] PHP-FPM pool configuration now can be changed

#### Fixed

- [#1381] resolved (either fixed or ignored) all the errors reported by phpstan level 4
- [#1495] Fixed missing zalas/phpunit-injector into project-base, Thanks to [@s3tezsky]
- [#1346] fix wrong annotation in Administrator
- [#1477] acceptance tests fixes
- [#1479] fix functional tests for single domain
- [#1336] fix typo (submited -> submitted), Thanks to [@sspooky13]
- [#1488] fixed usage of sed command on macos, Thanks to [@s3tezsky]
- [#1534] fixed importing css files for styleguide, Thanks to [@s3tezsky]

#### Unknown Category

- [#1499] - upgrade notes for lazyload, Thanks to [@TomasGottvald]
- [#1392] Hack phpunit to make tests clean
- [#1464] webline layout, Thanks to [@TomasGottvald]
- [#1313] force locale for number formatting, Thanks to [@pk16011990]
- [#1285] redirect logged user from registration page to homepage, Thanks to [@pk16011990]

### [shopsys/shopsys]

#### Added

- [#1310] guidelines-for-pull-request.md: added note about proper PR targeting
- [#1343] Added support to display dates in different timezone
- [#1305] BC promise: added info about Twig macros
- [#1281] Info about backend API installation added to 'Start you project' guide
- [#1252] faq: added tip for emails debugging
- [#1251] docs: added quick guide on how to see the progress of acceptance tests when using Docker
- [#1484] Added reference to bushman.cz into readme, Thanks to [@pesektomas]
- [#1451] Add annotation based configuration for smoke tests, Thanks to [@mariusbuescher]
- [#1485] Added styleguide to project
- [#1393] added error ID to 500 error page

#### Changed

- [#1270] improve upgrade notes for 8.0.0

#### Fixed

- [#1400] docs: installation guides: fixed formatting
- [#1421] fixed build.xml.patch
- [#1469] fixed wrongly resolved conflicts when 8.0 branch was merged to master
- [#1379] "annotations-check" and "annotations-fix" are turned off by default for existing projects to avoid BC-breaks
- [#1506] fixed version of symfony/monolog-bundle, Thanks to [@s3tezsky]
- [#1344] automated fixes and additions of annotations for extended classes

#### Removed

- [#1276] docker-compose-win.yml.dist: remove php.ini volume mount
- [#1329] remove notes about committing migrations-lock.yml in documentation
- [#1382] Removed info about not extensible admin forms in docs
- [#1416] remove docker-sync version from install guide, Thanks to [@sspooky13]
- [#1461] remove old IE support, Thanks to [@TomasGottvald]

#### Unknown Category

- [#1299] Updated license information in readme file
- [#1476] Unify working with images, Thanks to [@malyMiso]
- [#1289] composer.json enhancements (alphabetical sorting of extensions, 4 spaces as indentation), Thanks to [@sspooky13]
- [#1298] Update list of references in readme file
- [#1408] Dockerfile: stop defining default env vars
- [#1425] Eased installation of project with different domains and locales settings
- [#1308] domain configuration is autoloaded in build.xml
- [#1309] domain configuration is autoloaded in build.xml
- [#1319] make promo codes editable on separate pages
- [#1320] simplification of Travis CI configuration
- [#1532] made image lazy loading configurable, Thanks to [@s3tezsky]
- [#1354] upgrade the Adminer Docker image to 4.7
- [#1384] simplified filling variables in Kubernetes manifests, Thanks to [@s3tezsky]
- [#1352] docs: explain forced version of PHP in config.platform.php

## [v8.0.0] - 2019-07-30

### Unknown Package

- [#1265] Release of 8.0.0

<!-- dumped content end -->
```

File: upgrade/UPGRADE-v8.1.0.md [hunk 1]
```diff
@@ -258,6 +314,7 @@ There you can find links to upgrade notes for other versions too.
     - run `php phing annotations-fix` to fix or add all the relevant annotations for your extended classes
     - thanks to the fixes, your IDE (PHPStorm) will understand your code better
     - you can read more about the whole topic in the ["Framework extensibility" article](../introduction/framework-extensibility.md#making-the-static-analysis-understand-the-extended-code)
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v8.1.0.md]: This should be linked to docs.shopsys.com

File: upgrade/UPGRADE-v8.1.0.md [hunk 2]
```diff
@@ -266,11 +323,14 @@ There you can find links to upgrade notes for other versions too.
     - a lot of the possible issues should be already resolved if you followed the previous instruction and ran the `php phing annotations-fix` phing command
     - some of the issues related to class extension need to be addressed manually nevertheless (see the ["Framework extensibility" article](../introduction/framework-extensibility.md#problem-3)) for more information
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v8.1.0.md]: This should be linked to docs.shopsys.com

File: upgrade/UPGRADE-v8.1.0.md [hunk 3]
```diff
@@ -296,21 +356,25 @@ There you can find links to upgrade notes for other versions too.
...
         +       arguments:
         +           $useInlineEditation: false
         ```
+
 - add `setlocale(LC_NUMERIC, 'en_US.utf8');` in your `Bootstrap.php` right behind `setlocale(LC_CTYPE, 'en_US.utf8');` ([#1313](https://github.com/shopsys/shopsys/pull/1313/))
-    - add [`\Tests\ShopBundle\Unit\NumberFormattingTest`](https://github.com/shopsys/shopsys/blob/master/project-base/tests/ShopBundle/Unit/NumberFormattingTest.php)
+    - add [`\Tests\ShopBundle\Unit\NumberFormattingTest`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/tests/ShopBundle/Unit/NumberFormattingTest.php)
+
 - add support to display date-time values in different timezone ([#1343](https://github.com/shopsys/shopsys/pull/1343))
     - you can read more about how to [work with display timezone in documentation](/docs/introduction/working-with-date-time-values.md)
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v8.1.0.md]: This should be linked to docs.shopsys.com

File: upgrade/UPGRADE-v8.1.0.md [hunk 4]
```diff
@@ -492,6 +556,7 @@ There you can find links to upgrade notes for other versions too.
 - improve functional and smoke tests to be more readable and easier to write [#1392](https://github.com/shopsys/shopsys/pull/1392)
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v8.1.0.md]:
```suggestion
- improve functional and smoke tests to be more readable and easier to write ([#1392](https://github.com/shopsys/shopsys/pull/1392))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -29,6 +29,88 @@ The changelog is generated during the release process using [ChangelogLinker](ht
...
+- [#1464] webline layout
+
+#### Fixed
+
+- [#1285] redirect logged user from registration page to homepage, Thanks to [@pk16011990]
+- [#1313] force locale for number formatting, Thanks to [@pk16011990]
+- [#1336] fix typo (submited -> submitted), Thanks to [@sspooky13]
+- [#1346] fix wrong annotation in Administrator
+- [#1381] resolved (either fixed or ignored) all the errors reported by phpstan level 4
+- [#1479] fix functional tests for single domain
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply CHANGELOG.md]: I am not sure about it. Something in me says that single domain tests were failing even before. That is why I left it there.

File: upgrade/UPGRADE-v8.1.0.md [hunk 6]
```diff
@@ -17,27 +19,34 @@ There you can find links to upgrade notes for other versions too.
...
+    
+        +   # copy FPM pool configuration
+        +   COPY ${project_root}/docker/php-fpm/production-www.conf /usr/local/etc/php-fpm.d/www.conf
+    
+            COPY --chown=www-data:www-data / /var/www/html
+        ```
 
-        COPY --chown=www-data:www-data / /var/www/html
-    ```
 - production environment disallow administrators to log in with default credentials [#1360](https://github.com/shopsys/shopsys/pull/1360)
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply upgrade/UPGRADE-v8.1.0.md]: I do not understand this one. I see no space there 😕

File: upgrade/UPGRADE-v8.1.0.md (hunk 4 above)
Comment (s3tezsky) [reply upgrade/UPGRADE-v8.1.0.md]: Same here. I see no space at the beggining.

File: CHANGELOG.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply CHANGELOG.md]: 👌

**Pull request merged**